src/gossip/service.zig

```diff
@@ -1208,10 +1211,9 @@ pub const GossipService = struct {
             defer gossip_table_lock.unlock();
             var gossip_table: *GossipTable = gossip_table_lock.mut();
 
-            for (pull_requests.items) |*req| {
-                const caller = req.value.id();
+            for (pull_requests.items) |req| {
```

Comment:
should keep this as a pointer in this for loop (`*pull_request`) - without the pointer it will likely lead to a stack copy which we dont need since we only access the fields